plugins/zns/zns.c

```diff
@@ -223,7 +223,20 @@ static int zone_mgmt_send(int argc, char **argv, struct command *cmd, struct plu
 	if (fd < 0)
 		return errno;
 
-	if (cfg.data_len) {
+	if (!cfg.zsa) {
+		fprintf(stderr, "zone send action must be specified\n");
+		err = -EINVAL;
+		goto close_fd;
+	}
+
+	if (cfg.zsa == 0x10){
+		if(!cfg.data_len){
+			cfg.data_len = get_zdes_bytes(fd, cfg.namespace_id);
```

Comment:
If the data_len is not there, let's exit. The user should know that it supplies the data and the length of the data. What could happen, would be that a user thought it was 64bytes, but the data file provided was only 32 bytes, then the 32bytes after are essential unmapped, and would be garbled information that would be added to the zone.

Might be good to check the file size as well - but I am not sure if there is a precedence for that in nvme-cli. It is a bit on the defensive programming side.

Reply:
@MatiasBjorling I think its better to let it be like this. because I develop based on nvme-cli as is. for example, write support stdin input for data & metadata. and It can issue smaller data then data size & meta data size.
So I did same things for extended descriptor.